backend/src/src-licenses/src/main/java/com/siemens/sw360/licenses/db/LicenseDatabaseHandler.java

```diff
@@ -370,7 +372,10 @@ public RequestStatus updateLicense(License inputLicense, User user, User request
 
             License dbLicense = new License();
             if(inputLicense.isSetId()) {
-                dbLicense = licenseRepository.get(inputLicense.getId());
+                License dbLicenseTmp = licenseRepository.get(inputLicense.getId());
```

Comment:
I'd prefer that the license id stays optional - that's consistent with other thrift objects and their handling would be the same (i.e. backend decides what the id is)

As I see it, the presence of id in inputLicense is an indication that an update rather than add is being performed (for some reason, license service does not have separate methods for these actions.) So, if an update is requested and there's no license with that id in the db yet, then that's a license (pardon the pun) to throw an exception. I think an `assertNotNull` like at ComponentDatabaseHandler:401 would be well placed here. What do you think?